lib/open_api_spex/plug/put_api_spec.ex

```diff
@@ -48,4 +44,28 @@ defmodule OpenApiSpex.Plug.PutApiSpec do
     |> Stream.map(fn operation -> {operation.operationId, operation} end)
     |> Enum.into(%{})
   end
-end
\ No newline at end of file
+
+  defp fetch_spec(spec_module) do
+    # This if statement and the contents of the else block can be
+    # removed once an Elixir lang release requires an OTP version >=21.2
+    if function_exported?(:persistent_term, :info, 0) do
```

Comment:
I don't know how much performance impact this contributes, but I think there's a good chance you can make a compile-time conditional instead. The current OTP version is probably available at compile time.

Reply:
It's unclear if this is possible given that an application cannot be guaranteed to be deployed to an env with the same runtime as it was compiled in. Michal and others did not know if it would be guaranteed to work (or work well) in that situation.

I reworked the API to only check once, but as a general piece of info this check is very cheap.

Reply:
Yeah, that's a good point 👍